File: common/src/main/java/com/jon/common/presets/ProtocolConverter.java
package com.jon.common.presets;

import androidx.room.TypeConverter;

import com.jon.common.utils.Protocol;

class ProtocolConverter {
    @TypeConverter
    public static Protocol stringToProtocol(String string) {
        return Protocol.fromString(string);
    }

    @TypeConverter
    public static String protocolToString(Protocol protocol) {
        return protocol.get();
    }
}

File: common/src/main/java/com/jon/common/presets/ProtocolConverter.kt
package com.jon.common.presets

import androidx.room.TypeConverter
import com.jon.common.utils.Protocol

class ProtocolConverter {
    @TypeConverter
    fun stringToProtocol(string: String): Protocol {
        return Protocol.fromString(string)
    }

    @TypeConverter
    fun protocolToString(protocol: Protocol): String {
        return protocol.toString()
    }
}
